File: crash-course-python/sauce/sauce_example.py
# Selenium 3.14+ doesn't enable certificate checking
import os
import urllib3
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
from selenium import webdriver

sauce_username = os.environ["SAUCE_USERNAME"]
sauce_access_key = os.environ["SAUCE_ACCESS_KEY"]
remote_url = "https://ondemand.saucelabs.com/wd/hub"
capabilities = {
  'browserName': 'firefox',
  'browserVersion': '66.0',
  'platformName': 'Windows 10',
  'sauce:options': {
    'screenResolution': '1152x864',
    'username': sauce_username,
    'accessKey': sauce_access_key,
    'name': 'Testing Sauce Demo Site',
    'build': 'Sample Selenium-Python Test'
  }
}
driver = webdriver.Remote(command_executor=remote_url, desired_capabilities=capabilities)
driver.get("https://www.saucedemo.com")

driver.find_element_by_id('user-name').send_keys('standard_user')
driver.find_element_by_id('password').send_keys('secret_sauce')
driver.find_element_by_css_selector("[type='submit']").click()

actual_result = "passed" if driver.current_url == "https://www.saucedemo.com/inventory.html" else "failed"
driver.execute_script("sauce:job-result=" + actual_result)
driver.quit()

File: crash-course-ruby/sauce/sauce_example.rb
require 'selenium-webdriver'

#
# 1. Find username and access key here: https://app.saucelabs.com/user-settings and export them as Environment Variables
# for more info visit the documentation: https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Use+Environment+Variables+for+Authentication+Credentials
#
sauce_username = ENV['SAUCE_USERNAME']
sauce_access_key = ENV['SAUCE_ACCESS_KEY']

sauce_url = "https://@ondemand.saucelabs.com:443/wd/hub"

#
# 2. Grab browser and OS combos in capabilities from: https://wiki.saucelabs.com/display/DOCS/Platform+Configurator
#
capabilities = {
    browser_name: '',
    browser_version: '',
    platform_name: '',
    "sauce:options" => {
        username: sauce_username,
        accessKey: sauce_access_key,
        screen_resolution: '1152x864',
        name: "Testing Sauce Demo Site",
        build: "Sample Selenium-Ruby Test "
    }
}

@driver = Selenium::WebDriver.for :remote,
                                  url: sauce_url,
                                  desired_capabilities: capabilities

@driver.get "https://www.saucedemo.com/"

@driver.find_element(id: 'user-name').send_keys('standard_user')
@driver.find_element(id: 'password').send_keys('secret_sauce')
@driver.find_element(css: "[type='submit']").click

actual_result = @driver.current_url == "https://www.saucedemo.com/inventory.html" ? "passed" : "failed"
@driver.execute_script("sauce:job-result=#{actual_result}")
@driver.quit
